Lesson 2

aphmelev opened the pull request.

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
start

Commit:
изменение в задаче 1 алгоритм 3: if lst_copy[i] == lst_copy[i+1]:     # O(1)

Commit:
изменение в задаче 2 алгоритм 1: изменил на 2 цикла, теперь квадратичная

Commit:
изменение в задаче 3 алгоритм top3_company_v1: изменил сложности, но не понимаю почему for el in dict_list[:3]: - константаня

Commit:
изменение в задаче 4
def log_in_v1(users_list=users): - сложность изменена на квадратичную
if users_list.get(username): изменена на константаня

Commit:
изменение в задаче 5
добавлены методы удаления стопки, просмотр последнего елемента (количества тарелок) в последней стопке и вывод информации о текущем состоянии стопок и стеков

Commit:
Задача 1

Commit:
Задача 2

Commit:
Задача 3

Commit:
Задача 4

Commit:
Задача 5

Commit:
Задача 5

Commit:
Задача 6

Commit:
Задача 6 доработка проверки на число

Commit:
Задача 6 доработка проверки на число

Commit:
Задача 7

Commit:
Задача 4 доработка на проверку числа

Commit:
Коммит для создания новой ветки

DmitryChitalov commented on Урок 2. Практическое задание/task_2.py:
```diff
@@ -17,3 +17,22 @@
 Введите число: 123
 Количество четных и нечетных цифр в числе равно: (1, 2)
 """
+
```
нет решения задачи 1

DmitryChitalov commented on Урок 2. Практическое задание/task_2.py:
```diff
@@ -17,3 +17,22 @@
+        print("Введено не целое число, попробуйте еще раз!!!")
+
+
+find_number(input("Введите целое число: "))
```
try:
 except TypeError:

не должно быть в рекурсии
эту валидацию нужно вынести вне ф-ции

DmitryChitalov commented on Урок 2. Практическое задание/task_3.py:
```diff
@@ -18,3 +18,20 @@
+так как поздно принялся за решение задачи пока не понял как проверить что если значение
+заканчивается 0 или множеством 0 то перенести их на последние символы в перевернутом значении
+
+"""
```
выполнено

DmitryChitalov commented on Урок 2. Практическое задание/task_4.py:
```diff
@@ -10,3 +10,17 @@
+    input_number = int(input('Введите количество элементов: '))
+    print(f'Количество элементов - {input_number}, их сумма - {sum_of_elements(input_number,input_number)}')
+except ValueError:
+    print('Вы ввели строку, введите число!!!')
```
выполнено

DmitryChitalov commented on Урок 2. Практическое задание/task_5.py:
```diff
@@ -19,3 +20,16 @@
+        table_ascii_symbol(code + 1, count + 1)
+
+
+table_ascii_symbol()
```
выполнено

DmitryChitalov commented on Урок 2. Практическое задание/task_6.py:
```diff
@@ -7,3 +7,26 @@
+    except ValueError:
+        print('Вы ввели строку, введите число')
+        good_number()
+
```
тот же вопрос с try except

DmitryChitalov commented on Урок 2. Практическое задание/task_7.py:
```diff
@@ -12,3 +12,17 @@
+    if proof(number) == number * (number + 1) / 2:
+        print(f'{proof(number)} = {int(number * (number + 1) / 2)} - поздравляю, вы доказали равенство!!!')
+except ValueError:
+    print('Вы ввели строку, введите число!!!')
```
выполнено

Commit:
Задача 1 (очень странно, но она была в коммите)

Commit:
Задача 2
try:
except TypeError:
вынесено вне рекурсии

Commit:
Задача 6
try:
except TypeError:
опустил ниже базового случая
удалил рекурсивный запуск после except так как приводил к сбросу попыток, но теперь если ввести не число, скрипт будет остановлен, что тоже не айс, можно подшаманить, но, честно говоря пока нет времени

Commit:
Задача 6
try:
except TypeError:
опустил ниже базового случая
удалил рекурсивный запуск после except так как приводил к сбросу попыток, но теперь если ввести не число, скрипт будет остановлен, что тоже не айс, можно подшаманить, но, честно говоря пока нет времени
